m/reading_tools/sx2mat.m

```diff
@@ -0,0 +1,205 @@
+    end
+end
+
+
+%% Set option flags and values.
+output_format = lower(options.format);
+sensor_filtering = true;
+if ischar(options.variables) && strcmp(options.variables, 'all')
+    sensor_filtering = false;
+end
+sensor_list = cellstr(options.variables);
+a_cfg_file = ~strcmp(options.cfgfile, 'none'); % -FC
+
+
+%% Open the file.
+[fid, fid_msg] = fopen(filename, 'rt');
+if fid < 0
+    error('glider_toolbox:sx2mat:FileError', fid_msg);
+end
+
+%% Process the file.
+try
+    % read header and fill variables
+    header=fgetl(fid);
+    sensor_values = strread(header,'%s','delimiter',';');
+
+    [~, name, ext] = fileparts(filename);
+    meta.sources = {[name ext]};
+    meta.headers = {};
+    meta.variables = sensor_values;
+    meta.headers.variables_per_cycle = numel(sensor_values);
+    meta.headers.filename = {filename};
+    meta.units = repmat({'N/A'}, meta.headers.variables_per_cycle, 1);  % <------- should be fixed
+    meta.bytes = repmat({'N/A'}, meta.headers.variables_per_cycle, 1);  % <------- should be fixed
+
+    % read config file (specific to SeaExplorer)
+    if a_cfg_file % -FC
+        config = sxcfg(options.cfgfile);
+        meta.configuration_file = config;
+    end
+    
+    % fill 'data'
+    % Read sensor data filtering selected variables if needed.
+    sensor_format = repmat({'%f'}, meta.headers.variables_per_cycle, 1);
+    if sensor_filtering
+        selected_variables = ismember(meta.variables, sensor_list);
+        meta.variables = meta.variables(selected_variables);
+        meta.units = meta.units(selected_variables);
+        meta.bytes = meta.bytes(selected_variables);
+        sensor_format(~selected_variables) = {'%*f'};
+    end
+
+    fmt_str = [sprintf('%s ', sensor_format{1:end-1}) sensor_format{end} '\n'];
+    fmt_str(1:2) = '%s';
+    data_values = textscan(fid, fmt_str, 'ReturnOnError', false, 'headerlines', 1, 'delimiter', ';');
+
+    % transform time stamp in posix vector -FC
+    Timestamp = data_values{1};
+    if length(Timestamp{1}) == 19 % Nav files
+        data_values{1} = (datenum(Timestamp, 'dd/mm/yyyy HH:MM:SS') ...
+                          - datenum(1970,1,1))*86400;
+    elseif length(Timestamp{1}) == 23 % Nav files
+        data_values{1} = (datenum(Timestamp, 'dd/mm/yyyy HH:MM:SS.FFF') ...
+                          - datenum(1970,1,1))*86400;
+    else
+        error('Timestamp format unknown!');        
+    end
+        
+    meta.variables{1} = 'Posixtime';
+    data = [data_values{:}];
+    
+    switch output_format
+      case 'array'
+        % nothing to do, done just above -FC
+      case 'struct' % <============================== to be tested!
+        data = cell2struct(data_values, meta.variables, 2);
+      otherwise
+        error('glider_toolbox:sx2mat:InvalidFormat', ...
+              'Invalid output format: %s.', format)
+    end
+
+catch exception
+    % Close the file after a reading error.
+    fclose(fid);
+    rethrow(exception);
+end
```

Comment:
Indentation is broken in this file (the toolbox uses 2 spaces per level),
and a final `end` should close the function definition for coherence with the coding style of the toolbox
(this is cosmetic, the interpreter allows the alternate syntax)